lib/bugsnag/logger_backend.ex

```diff
@@ -0,0 +1,43 @@
+defmodule Bugsnag.LoggerBackend do
+  require Logger
+
+  @behaviour :gen_event
+
+  @impl :gen_event
+  def init(__MODULE__) do
+    {:ok, []}
+  end
+
+  @impl :gen_event
+  def handle_call({:configure, new_options}, options) do
+    {:ok, :ok, Keyword.merge(options, new_options)}
+  end
+
+  @impl :gen_event
+  def handle_event({_level, _group_leader, {Logger, _message, _timestamp, metadata}}, options) do
+    try do
+      case metadata[:crash_reason] do
+        {{:nocatch, _term}, _stacktrace} ->
+          {:ok, options}
+
+        {%{__exception__: _module} = exception, stacktrace} ->
+          Bugsnag.sync_report(exception, stacktrace: stacktrace)
+          {:ok, options}
+
+        {_exit_value, _stacktrace} ->
+          {:ok, options}
+
+        nil ->
+          {:ok, options}
```

Comment:
We've been using a module similar to this internally for the last couple years.  It'd be great to see it in Bugsnag!

We report errors from the other branches of the `case` as well (to track any process crashes, or uncaught `throw`s) as well.  I've copy/pasted (with a few minor edits) our module if you want some inspiration: https://gist.github.com/darrenclark/971770561cc125e328decadabba8f817